File: C/宝くじ.py
# 入力を受け取る
b = int(input())
n = int(input())
a = [int(input()) for _ in range(n)]

for i in range(n):
    if b == a[i]: # １等であるかどうか
        print("first")
    elif (a[i] + 1 == b) or (a[i] - 1 == b): # 前後賞であるかどうか
        print("adjacent")
    elif b % 10000 == a[i] % 10000: # ２等かどうか
        print("second")
    elif b % 1000 == a[i] % 1000: # ３等かどうか
        print("third")
    else: # ハズレ
        print("blank")

File: C++/C/宝くじ.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    // 入力を受け取る
    int b, n;
    cin >> b >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    for (int i = 0; i < n; i++) {
        if (b == a[i]) { // １等であるかどうか
            cout << "first" << endl;
        } else if ((a[i] + 1 == b) || (a[i] - 1 == b)) { // 前後賞であるかどうか
            cout << "adjacent" << endl;
        } else if (b % 10000 == a[i] % 10000) { // ２等かどうか
            cout << "second" << endl;
        } else if (b % 1000 == a[i] % 1000) { // ３等かどうか
            cout << "third" << endl;
        } else { // ハズレ
            cout << "blank" << endl;
        }
    }

    return 0;
}